File: Ruby/Jenny's secret message.rb
https://www.codewars.com/kata/55225023e1be1ec8bc000390

def greet(name)
    return "Hello, my love!" if name == "Johnny"
    return "Hello, #{name}!"
  end

/Test Cases:/

describe "Jenny's greeting function" do
    it "should greet some people normally" do
      Test.assert_equals greet("James"), "Hello, James!"
      Test.assert_equals greet("Jane"), "Hello, Jane!"
      Test.assert_equals greet("Jim"), "Hello, Jim!"
    end
    it "should greet Johnny a little bit more special" do
      Test.assert_equals greet("Johnny"), "Hello, my love!"
    end
  end

File: C#/Jenny's Secret message.cs
//https://www.codewars.com/kata/55225023e1be1ec8bc000390

using System;

public static class Kata
{
  public static string greet(string name)
  {

    if(name == "Johnny")
      return "Hello, my love!";
    return "Hello, " + name + "!";
  }
}

//Test Cases

[TestFixture]
public class JennysGreeting
{
  [Test]
  public static void ShouldGreetJimNormally()
  {
    Assert.AreEqual("Hello, Jim!", Kata.greet("Jim"));
  }
  [Test]
  public static void ShouldGreetJaneNormally()
  {
    Assert.AreEqual("Hello, Jane!", Kata.greet("Jane"));
  }
  [Test]
  public static void ShouldGreetSimonNormally()
  {
    Assert.AreEqual("Hello, Simon!", Kata.greet("Simon"));
  }
  
  [Test]
  public static void ShouldGreetJohnnySpecially()
  {
    Assert.AreEqual("Hello, my love!", Kata.greet("Johnny"));
  }
}